datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -491,77 +436,33 @@ impl FileOpener for ParquetOpener {
-                        for predicate in page_index_predicates {
-                            // `extract_page_index_push_down_predicates` only return predicate with one col.
-                            let col_id = *predicate
-                                .need_input_columns_ids()
-                                .iter()
-                                .next()
-                                .unwrap();
-                            let mut selectors = Vec::with_capacity(row_groups.len());
-                            for r in &row_groups {
-                                let rg_offset_indexes = file_offset_indexes.get(*r);
-                                let rg_page_indexes = file_page_indexes.get(*r);
-                                if let (Some(rg_page_indexes), Some(rg_offset_indexes)) =
-                                    (rg_page_indexes, rg_offset_indexes)
-                                {
-                                    selectors.extend(
-                                        prune_pages_in_one_row_group(
-                                            &groups[*r],
-                                            &predicate,
-                                            rg_offset_indexes.get(col_id),
-                                            rg_page_indexes.get(col_id),
-                                            &file_metrics,
-                                        )
-                                            .map_err(|e| {
-                                                ArrowError::ParquetError(format!(
-                                                    "Fail in prune_pages_in_one_row_group: {}",
-                                                    e
-                                                ))
-                                            }),
-                                    );
-                                } else {
-                                    // fallback select all rows
-                                    let all_selected = vec![RowSelector::select(
-                                        groups[*r].num_rows() as usize,
-                                    )];
-                                    selectors.push(all_selected);
-                                }
-                            }
-                            debug!(
-                            "Use filter and page index create RowSelection {:?} from predicate:{:?}",
-                            &selectors, predicate
-                        );
-                            row_selections.push_back(
-                                selectors.into_iter().flatten().collect::<Vec<_>>(),
-                            );
-                        }
-                        let final_selection = combine_multi_col_selection(row_selections);
-                        builder = builder.with_row_selection(final_selection.into());
-                    }
-                }
+            // page index pruning: if all data on individual pages can
```

Comment:
The core change of this PR is to refactor `build_page_filter` into a function and move its code into the `page_filter` module

Reply:
👍